src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -1125,12 +1130,13 @@ protected async Task<ImmutableArray<(Project project, Diagnostic diagnostic)>> G
             foreach (var project in solution.Projects)
             {
                 var (compilation, generatorDiagnostics) = await GetProjectCompilationAsync(project, verifier, cancellationToken).ConfigureAwait(false);
-                var compilationWithAnalyzers = CreateCompilationWithAnalyzers(compilation, analyzers, GetAnalyzerOptions(project), cancellationToken);
+                var analyzerOptions = GetAnalyzerOptions(project);
+                var compilationWithAnalyzers = CreateCompilationWithAnalyzers(compilation, analyzers, analyzerOptions, cancellationToken);
 
                 ImmutableArray<Diagnostic> allDiagnostics;
                 if (AnalysisResultWrapper.WrappedType is not null)
                 {
-                    var compilationDiagnostics = compilation.GetDiagnostics(cancellationToken);
```

Comment:
📝 This line was the source of the problem. It bypassed diagnostic suppressors when obtaining compiler diagnostics.